core/deployment/src/main/java/io/quarkus/deployment/builditem/ExtraIndexBuildItem.java

```diff
@@ -0,0 +1,18 @@
+package io.quarkus.deployment.builditem;
+
+import org.jboss.jandex.IndexView;
+
+import io.quarkus.builder.item.SimpleBuildItem;
+
+public final class ExtraIndexBuildItem extends SimpleBuildItem {
```

Comment:
In theory, you may need to register other kinds of generated classes, not only beans. Therefore, it might be reasonable to extend `MultiBuildItem`. I'd also recommend to rename the build item, maybe something like `GeneratedClassesIndexBuildItem`, plus add some basic javadoc that explains the reasoning...

Reply:
@mkouba class renamed, but can't use `MultiBuildItem` as it causes errors during start

```
Caused by: java.lang.IllegalArgumentException: Unsupported method parameter class io.quarkus.deployment.builditem.GeneratedClassesIndexBuildItem at io.quarkus.deployment.builditem.GeneratedClassesIndexBuildItem generatedClassesIndexBuildItem of public void io.quarkus.deployment.steps.ReflectiveHierarchyStep.build(io.quarkus.deployment.builditem.CombinedIndexBuildItem,io.quarkus.deployment.builditem.GeneratedClassesIndexBuildItem,java.util.List,java.util.List,java.util.List,io.quarkus.deployment.annotations.BuildProducer) throws java.lang.Exception of class io.quarkus.deployment.steps.ReflectiveHierarchyStep
	at io.quarkus.deployment.ExtensionLoader.reportError(ExtensionLoader.java:979)
	at io.quarkus.deployment.ExtensionLoader.loadStepsFrom(ExtensionLoader.java:769)
	at io.quarkus.deployment.ExtensionLoader.loadStepsFrom(ExtensionLoader.java:211)
```

not sure why ....